qcodes/instrument_drivers/Keysight/keysight_34980a_submodules.py

```diff
@@ -0,0 +1,185 @@
+from qcodes import VisaInstrument, InstrumentChannel
+from typing import Union, List, Tuple, Optional
+
+
+class KeysightSubModule(InstrumentChannel):
+    """
+    A base class for submodules for the 34980A systems.
+
+    Args:
+        parent: the system which the module is installed on
+        name (str): user defined name for the module
+        slot (int): the slot the module is installed
+    """
+    def __init__(
+            self,
+            parent: Union[VisaInstrument, InstrumentChannel],
+            name: str,
+            slot: int
+    ) -> None:
+
+        super().__init__(parent, name)
+
+        self.slot = slot
+
+
+class KeysightSwitchMatrixSubModule(KeysightSubModule):
+    """
+    A base class for **Switch Matrix** submodules for the 34980A systems.
+
+    Args:
+        parent: the system which the module is installed on
+        name (str): user defined name for the module
+        slot (int): the slot the module is installed
```

Comment:
there's no need to use type annotations in the docstring if the method signature already has type annotations. hence please remove the types from the docstring (this comment applies to the whole PR :) )

Reply:
Done, removed all duplicated type annotations in the docstring if the method already has type annotations.